src/replay_stage.rs

```diff
@@ -317,6 +332,12 @@ impl ReplayStage {
                         // We've reached the end of a slot, reset our state and check
                         // for leader rotation
                         if max_tick_height_for_slot == current_tick_height {
+                            let bank_id = current_slot.unwrap();
```

Comment:
I think this bank_id could potentially be the same as the bank we set in the beginning of the function at line 221, causing a bank overwrite on lines 354-355